OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/Analytics.java

```diff
@@ -841,6 +841,7 @@ public static String getShareTypeValue(@NonNull ShareUtils.ShareType shareType)
         public static String removeUnSupportedCharacters(String value) {
             return value.replaceAll(":", "_")
                     .replaceAll("-", "_")
+                    .replaceAll("_ ", "_")
```

Comment:
Some suggestions/improvements:
1. How about using StringBuilder instead of performing the replaceAll again and again on the String value?
2. Why don't we just replace the space character with underscore here instead (i.e. ` .replaceAll(" ", "_")`)? (cuz at the end double underscore is eventually going to be replaced with a single one)